angular/src/directives/CipherListVirtualScroll.directive.ts

```diff
@@ -0,0 +1,101 @@
+import {
+    coerceNumberProperty,
+    NumberInput,
+} from '@angular/cdk/coercion';
+import {
+    FixedSizeVirtualScrollStrategy,
+    VIRTUAL_SCROLL_STRATEGY,
+} from '@angular/cdk/scrolling';
+import {
+    Directive,
+    forwardRef,
+    Input,
+    OnChanges,
+} from '@angular/core';
+
+// Custom virtual scroll strategy for cdk-virtual-scroll
+// Uses a sample list item to set the itemSize for FixedSizeVirtualScrollStrategy
+// The use case is the same as FixedSizeVirtualScrollStrategy, but it avoids locking in pixel sizes in the template.
+export class CipherListVirtualScrollStrategy extends FixedSizeVirtualScrollStrategy {
+    private checkItemSizeCallback: any;
+    private timeout: any;
+
+    constructor(itemSize: number, minBufferPx: number, maxBufferPx: number, checkItemSizeCallback: any) {
+        super(itemSize, minBufferPx, maxBufferPx);
+        this.checkItemSizeCallback = checkItemSizeCallback;
+    }
+
+    onContentRendered() {
+        if (this.timeout != null) {
+            clearTimeout(this.timeout);
+        }
+
+        this.timeout = setTimeout(this.checkItemSizeCallback, 500);
+    }
+}
+
+export function _cipherListVirtualScrollStrategyFactory(cipherListDir: CipherListVirtualScroll) {
+    return cipherListDir._scrollStrategy;
+}
+
+@Directive({
+    selector: 'cdk-virtual-scroll-viewport[defaultItemSize]',
+    providers: [{
+        provide: VIRTUAL_SCROLL_STRATEGY,
+        useFactory: _cipherListVirtualScrollStrategyFactory,
+        deps: [forwardRef(() => CipherListVirtualScroll)],
+    }],
+})
+export class CipherListVirtualScroll implements OnChanges {
+    // tslint:disable
+    static ngAcceptInputType_defaultItemSize: NumberInput;
```

Comment:
does this just limit the types available to the scrolling strategy attributes? Isn't that already checked by typescript?